virtualizarr/vendor/zarr/utils.py

```diff
@@ -0,0 +1,22 @@
+import json
+import numbers
+
+from typing import Any
+
+
+class NumberEncoder(json.JSONEncoder):
+    def default(self, o):
+        # See json.JSONEncoder.default docstring for explanation
+        # This is necessary to encode numpy dtype
+        if isinstance(o, numbers.Integral):
+            return int(o)
+        if isinstance(o, numbers.Real):
+            return float(o)
+        return json.JSONEncoder.default(self, o)
+
+
+def json_dumps(o: Any) -> bytes:
```

Comment:
I chose to vendor this because I didn't want to import the zarr-python library when it's in flux, and also this helps make it clear exactly which parts of this package even need `zarr-python` at all.